lib/src/git.rs

```diff
@@ -902,8 +903,16 @@ impl<'a> RemoteCallbacks<'a> {
                         });
                     }
 
-                    if let Some(ref mut cb) = self.get_ssh_key {
-                        if let Some(path) = cb(username) {
+                    if ssh_key_paths_to_try.is_none() {
+                        if let Some(ref mut cb) = self.get_ssh_keys {
+                            let mut paths = cb(username);
+                            paths.reverse();
+                            ssh_key_paths_to_try = Some(paths);
+                        }
+                    }
+
+                    if let Some(paths) = &mut ssh_key_paths_to_try {
+                        if let Some(path) = paths.pop() {
```

Comment:
Does this mean we try only the first SSH key that exists?

Reply:
It's supposed to mean that we try one after another. Do you think it won't work or were you just curious?

Reply:
Oh - whoops, I completely misread this. For some reason I thought ssh_key_paths_to_try was being set up _inside_ the callback. I should really not be reading code at 2am.